plane/src/proxy/rewriter.rs

```diff
@@ -104,10 +104,11 @@ impl RequestRewriter {
             }
         };
 
-        // Remove port from hostname if present.
-        let hostname = match hostname.split_once(':') {
-            Some((hostname, _)) => hostname,
-            None => hostname,
+        // Remove port from hostname if present and port is 443.
```

Comment:
I think your explanation in the description about why 443 is really the only one we need to care about makes a lot of sense. Is it worth putting some of that in this comment, too?